src/rest-server/config/rest_server.py

```diff
@@ -58,6 +74,22 @@ def run(self):
             'schedule-port-start', 'schedule-port-end', 'sql-max-connection'
         ]:
             service_object_model[k] = self.service_configuration[k]
+
+        # default-computing-device-type only works in default scheduler.
+        # If hived scheduler is enabled, it will be ignored.
+        # It determines:
+        # (1) the resource name to be added in containers[*].resources.limits when we submit job.
+        # (2) how we count the gpu number.
+        # Users should place their computing device in `layout.yaml`.
+        # Currently we only support one computing device in default scheduler.
+        # Here we find all available computing devices from `layout.yaml`,
+        # and use the first one as the default computing device type.
```

Comment:
when changing `computing_device_types` from `set` to `list`, the order is undefined. It's probably not the one first added.
https://github.com/microsoft/pai/blob/8dbb5c3d04fa37c5a5529c0c208ea5bcb33e3024/src/rest-server/config/rest_server.py#L43
https://github.com/microsoft/pai/blob/8dbb5c3d04fa37c5a5529c0c208ea5bcb33e3024/src/rest-server/config/rest_server.py#L53-L54
https://github.com/microsoft/pai/blob/8dbb5c3d04fa37c5a5529c0c208ea5bcb33e3024/src/rest-server/config/rest_server.py#L89

Reply:
Done. Use another `list` together with `set`.